pkg/apis/serving/v1beta1/predictor.go

```diff
@@ -19,8 +19,15 @@ package v1beta1
 import (
 	"github.com/kubeflow/kfserving/pkg/constants"
 	v1 "k8s.io/api/core/v1"
+	"reflect"
 )
 
+// PredictorImplementation defines common functions for all predictors e.g Tensorflow, Triton, etc
+// +kubebuilder:object:generate=false
+type PredictorImplementation interface {
+	IsFrameworkSupported(framework string, config *InferenceServicesConfig) bool
```

Comment:
Is there any other functions are predictor specific?

Reply:
I do not think so. I am going to add a memory check but that seems to be relevant to both predictors and explainers.